### PR TITLE
Database/mgmt

### PR DESCRIPTION
**Pull request by Whatsuup**
## Backend
Make sure, you have done de following before merging this pull request:
- [x] Your code is **documented** using doc comments.
- [x] You added appropriate **logging** in your code.
- [x] You added **unit tests** to all your implemented functions.
- [x] You ran `cargo format` to format your code.
- [x] You ran `cargo clippy` to check for improvements.
- [x] You added the **class descriptions** of the classes you implemented in the **latex document** on overleaf.
      Make sure to mark any changes you made to the design documents specification.

### CODE REVIEW

File: backend/src/layer/logic/mealplan_management/relation_resolver.rs [hunk 1]
```diff
@@ -1,9 +1,32 @@
 use crate::interface::mensa_parser::model::{Dish, ParseCanteen, ParseLine};
 use crate::interface::persistent_data::{DataError, MealplanManagementDataAccess};
 use crate::util::{Date, Uuid};
+use std::collections::HashMap;
 use std::slice::Iter;
+use lazy_static::lazy_static;
 use tracing::warn;
 
+lazy_static! {
+    static ref CANTEEN_POSITIONS: HashMap<&'static str, u32> = HashMap::from([
```
Comment (worldofjoni) [inline backend/src/layer/logic/mealplan_management/relation_resolver.rs:10]: is it possible to not hard code thah but rather infer it from the to the parser provided list of canteens?
We then probably have to do that in the parse componennt

File: backend/src/layer/logic/mealplan_management/relation_resolver.rs [hunk 2]
```diff
@@ -33,29 +56,35 @@ where
...
+                self.db
+                    .insert_canteen(&canteen.name, Self::get_position(&canteen.name))
                     .await?
             }
-            None => self.db.insert_canteen(&canteen.name, todo!()).await?,
         };
         self.db.dissolve_relations(db_canteen, date).await?;
+        let mut position: u32 = 0;
         for line in canteen.lines {
+            position += 1;
```
Comment (worldofjoni) [inline backend/src/layer/logic/mealplan_management/relation_resolver.rs:75]:
```suggestion
        for (position, line) in canteen.lines.iter().enumerate() {
```

**Review by worldofjoni — CHANGES_REQUESTED** [review]

File: backend/src/layer/logic/mealplan_management/relation_resolver.rs (hunk 1 above)
Comment (Whatsuup) [reply backend/src/layer/logic/mealplan_management/relation_resolver.rs]: deleted

File: backend/src/layer/logic/mealplan_management/relation_resolver.rs (hunk 2 above)
Comment (Whatsuup) [reply backend/src/layer/logic/mealplan_management/relation_resolver.rs]: deleted, used in html_parser

**Review by worldofjoni — APPROVED** [review]

**Pull request merged**